### PR TITLE
refactor(auth): extract KnownAccountsRegistry — known-accounts CRUD + migration (#4741)

### PR DESCRIPTION
**Pull request by realmeylisdev**
## What & why

Third **repository-tier** slice of #4741 (after R1 #5757, R2 #5805). The known-accounts registry — the multi-account welcome-screen picker source — moves out of `auth_service.dart` behind the stable facade. auth_service.dart **4,776 → 4,442**.

## The stateless collaborator

New `lib/services/auth/known_accounts_registry.dart` (`KnownAccountsRegistry`). Unlike R2's `OAuthSessionCoordinator` this one is **STATELESS** — every method reads `SharedPreferences` fresh, so there's no cached state and no `detach()`/sign-out hook. It needs **no tear-off ports** either: just three injected deps (`SecureKeyStorage`, `SignerSecureStore`, `FlutterSecureStorage?`), and it imports `keycast_flutter` directly for the one `KeycastSession.load` in the migration path (the R2 precedent — the `auth/` collaborators already depend on keycast_flutter).

Moved verbatim into the collaborator:
- `getKnownAccounts()` — read + MRU sort + the `null → one-time migration` vs `empty → []` branch
- the one-time legacy migration (`_migrateLegacyAccount` + `_migrateAutomaticKeys` + `_persistMigrationResult`)
- `upsert` (was `_addToKnownAccounts`) and `remove` (was `_removeFromKnownAccounts`)
- the restorable-account filter (`restorableAccounts` + `_hasRestorableLocalKey` + `_hasRestorableSignerArchive`)
- **new** `persist(List<KnownAccount>)` so the registry owns **every** `kKnownAccountsKey` write (including the ones the facade's sign-out recovery used to do inline)

## Seam (what stays on the facade)

The facade keeps all session state and restore orchestration — nothing that reads `_authState`/`_authSource`/`currentPublicKeyHex` moves:
- `signInForAccount`, `_restoreLastUsedAccountOrFallback`, `_tryRestoreFromKnownAccounts` (the live `AccountRestoreFailedException` guard is facade-owned)
- `signOut` ordering invariants unchanged (anchor-write, archive/remove-before-teardown, reset-recovery-after-signer-cleanup)
- `_resetRecoveryAfterLocalAccountRemoval` keeps its `_authSource = none` / `authentication_source` / `last_used_npub` resets and its facade-layer logs; it now calls `_knownAccounts.restorableAccounts(...)` + `_knownAccounts.persist(...)` instead of the inline filter + `prefs.setString(kKnownAccountsKey, ...)`.

Public surface is preserved: `getKnownAccounts()` and `removeKnownAccount()` stay as thin delegators, so all 108 consumers are untouched.

## Log `name:` retag (not drift)

Every moved log line keeps its **message content byte-identical** vs `origin/main`; only the structured `name:` tag is retagged `AuthService` → `KnownAccountsRegistry` at the moved sites — the same R1/R2 own-class-tag convention. The only thing this affects is an external log filter keyed on `name:'AuthService'` for these specific lines (nothing in-repo keys on that tag). The facade-layer logs (in `signOut` / the reset method) stay `name:'AuthService'`.

## Verification

- **Adversarial drift-hunter (3 independent lenses vs `origin/main`): zero behavior drift.** Confirmed all five moved clusters are token-for-token identical (control flow, await ordering, the null-vs-empty branch, `length == 64` guards, the `!accounts.any` dedup, the `copyWith` upsert preserving `addedAt`, the `removeWhere`, the restorable switch arms). Confirmed the two wiring substitutions are transparent: `origin/main`'s `_loadAmberInfo()`/`_loadBunkerInfo()` were exact one-line delegators to `_signerStore.loadAmber()`/`loadBunker()`; and `persist(const <KnownAccount>[])` emits `"[]"`, byte-identical to the old `jsonEncode(<Object>[])`. All 12 call sites rewired with identical args; `dart:convert` import removal from the facade verified safe.
- **New `known_accounts_registry_test.dart` — 26 tests** covering getKnownAccounts (sort / sealed-empty / parse-failure), every migration arm (automatic / amber / bunker / divineOAuth / source=none, dedup, seal-once, error-swallow), upsert (insert + update-preserving-addedAt), remove, persist (overwrite + empty→`"[]"`), and restorableAccounts (local-key match, primary-key match, no-material drop, archive keep/drop, none/nip07 always dropped, verification-error → not restorable).
- **`multi_account` (91 tests) stays byte-identical** — zero edits. (Its group-name strings still say `_migrateLegacyAccount` / `_addToKnownAccounts`; those are pre-extraction labels left untouched on purpose, since editing the pinned suite would itself signal drift.) Pinned suites green: identity_change_preservation, destructive_signout, signout_cleanup, oauth_recovery, expired_session_downgrade.
- **375-test auth sweep green; `flutter analyze` clean; `dart format` clean.** CI ratchets pass: untested-services floor (new file ships its same-named test), empty-catch ceiling (no bare catch moved), service-sentinel (43→20, decreases are free).

No user-facing change. Part of #4741. Next: PR5–6 in-place split completion, then the `packages/auth_repository` scaffold.

### CODE REVIEW

**Review by hm21 — COMMENTED** [review]
Clean, faithful extraction. Verified locally: the five moved clusters are token-identical to `origin/main` (spot-checked `_migrateAutomaticKeys` byte-for-byte), the two wiring swaps are transparent (`_loadAmberInfo`/`_loadBunkerInfo` were exact one-line delegators on main and stay in use on four other paths — no dead code), and `persist(const <KnownAccount>[])` emits `"[]"`. Log `name:` retags are consistent (moved sites → `KnownAccountsRegistry`, facade-layer stays `AuthService`). Registry suite (26) + `multi_account` (91) green locally, `analyze` clean. Correct sequencing too — waited for R2 to merge, base is fresh `main`, not stacked.

One non-blocking nit inline.

File: mobile/lib/services/auth/known_accounts_registry.dart [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+import 'package:openvine/services/auth/signer_secure_store.dart';
+import 'package:openvine/utils/nostr_key_utils.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:unified_logger/unified_logger.dart';
+
+// Mirrors the facade's `authentication_source` SharedPreferences key; the
+// legacy migration reads it to recover the pre-multi-account session. Kept in
+// sync with `_kAuthSourceKey` in auth_service.dart — both pin the same literal,
+// which the multi_account suite asserts.
+const _kAuthSourceKey = 'authentication_source';
```
Comment (hm21) [inline mobile/lib/services/auth/known_accounts_registry.dart]: Nit (non-blocking): `_kAuthSourceKey = 'authentication_source'` is now duplicated as a private const in both the facade and here, guarded only by this comment + a literal-asserting test. `kKnownAccountsKey` already sets the precedent of hoisting a shared storage key into a public const in `models/known_account.dart` — promoting `authentication_source` there (e.g. `kAuthenticationSourceKey`) would remove the drift risk structurally instead of by convention. Fine to defer to a later slice if you'd rather keep this PR pure-move.

Comment (realmeylisdev) [reply mobile/lib/services/auth/known_accounts_registry.dart]: Done in 5e539df584 — hoisted to a single public `kAuthenticationSourceKey` and dropped both private `_kAuthSourceKey` copies, so the two call sites share one source of truth instead of a comment + literal-asserting test.

One deviation from the "e.g." you gave: I put it in `models/authentication_source.dart` rather than `known_account.dart`. The value stored under this key is an `AuthenticationSource.code`, so keeping the key next to its own model is the faithful read of the `kKnownAccountsKey`-lives-with-`KnownAccount` precedent you cited — and the facade already re-exports `authentication_source.dart`, so the const lands on the public surface for free. Happy to move it to `known_account.dart` if you'd rather keep all storage keys colocated.

Literal value is unchanged, so the `authentication_source` fixtures/assertions stay byte-identical. Registry (26) + multi_account (91) + the four fixture-asserting auth suites and `analyze` all green.

**Review by hm21 — APPROVED** [review]
LGTM
